File: src/main/java/dev/igoyek/eye/util/UserUtil.java
package dev.igoyek.eye.util;

import dev.igoyek.eye.config.AppConfig;
import net.dv8tion.jda.api.entities.Guild;
import net.dv8tion.jda.api.entities.User;

public final class UserUtil {

    private static AppConfig appConfig;

    public static boolean isUserBanned(User user) {
        Guild guild = user.getJDA().getGuildById(appConfig.guildId);

        if (guild == null) return false;

        return guild.retrieveBanList().complete().stream().anyMatch(ban -> ban.getUser().getId().equals(user.getId()));
    }
}

File: src/main/java/dev/igoyek/eye/util/UserUtil.kt
package dev.igoyek.eye.util

import dev.igoyek.eye.config.AppConfig
import net.dv8tion.jda.api.entities.User

class UserUtil {

    companion object {
        private var appConfig: AppConfig? = null

        @JvmStatic
        fun isUserBanned(user: User): Boolean {
            val guild = user.jda.getGuildById(appConfig?.guildId ?: return false) ?: return false

            return guild.retrieveBanList().complete().any() { it.user.id == user.id }
        }
    }
}